src/java/main/org/apache/zookeeper/jmx/MBeanRegistry.java

```diff
@@ -124,9 +124,7 @@ private void unregister(String path,ZKMBeanInfo bean) throws JMException  {
             return;
         if (!bean.isHidden()) {
             final ObjectName objName = makeObjectName(path, bean);
-            if (LOG.isInfoEnabled()) {
-                LOG.info("Unregister MBean [{}]", objName);
-            }
+            LOG.debug("Unregister MBean [{}]", objName);
```

Comment:
any idea why this was in an if clause before. i don't think it was needed, but just checking.

Reply:
this sounds an optimization as with the if clause presence, if the log is not enabled the objName is not copied which might help if copying objName is expensive. sounds not a very big deal (and i noticed we have two inconsistent logging style in code base, with or without this if clause, might worth a clean up).

Reply:
Right. And this where using log4j message formatter comes important: if one use `string.format()` instead and not putting it inside an if block, the string will be formatted every time no matter if the log level is enabled.